File: dtc.frontend/src/dtos/tournamentConfig/TournamentConfigDto.ts
import z from "zod";
import { TournamentMode } from "../../enums/TournamentMode";

export const tournamentConfigDtoSchema = z.object({
  id: z.number().int(),
  tournamentId: z.number().int(),
  versionNr: z.number().int(),
  mode: z.enum(TournamentMode),
  matchDurationMinutes: z.number().int(),
  breakBetweenMatchesMinutes: z.number().int(),
  groupCount: z.number().int(),
  playersPerGroup: z.number().int().nullable(),
  qualifiersPerGroup: z.number().int(),
});

export type TournamentConfigDto = z.infer<typeof tournamentConfigDtoSchema>;

File: DTC.Api/Dtos/TournamentConfig/TournamentConfigDto.cs
using DTC.Api.Enums;

namespace DTC.Api.Dtos.TournamentConfig
{
    public class TournamentConfigDto
    {
        public int Id { get; set; }

        public int TournamentId { get; set; }

        public TournamentMode Mode { get; set; }

        public int MatchDurationMinutes { get; set; }

        public int BreakBetweenMatchesMinutes { get; set; }

        public int GroupCount { get; set; }

        public int? PlayersPerGroup { get; set; }

        public int QualifiersPerGroup { get; set; }
    }
}
